### PR TITLE
docs(carbon-platform): index charts with schema

### PR DESCRIPTION
**Pull request by mattrosno**
### Updates
- Indexes these libraries and assets using the schema (https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md)
  - [x] Vanilla
  - [x] Angular
  - [x] React
  - [x] Svelte
  - [x] Vue 

### Demo screenshot or recording
N/A

### To do
- [x] Need thumbnails for
  - [x] Combo
  - [x] Sparkline
  - [x] Step
- [x] Need tags for (https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#data-visualization-tags)
  - [x] ~~Combo~~ (we don't need an additional tag for every asset if there's not a fit)
  - [x] Sparkline
  - [x] Step
- [x] Should `Meter (proportional)` be indexed? I see in as top-level in Storybook, but not listed in legacy website index. If so, we need a thumbnail for it.

### To do some time later

- [ ] Add descriptions to every component

### CODE REVIEW

File: packages/core/carbon.yml [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+      - geographic-overlay
+    framework: design-only
+    platform: web
+    thumbnailPath: /src/thumbnails/geo-choropleth.svg
+    externalDocsUrl: https://www.carbondesignsystem.com/data-visualization/complex-charts#proportional-symbol-map
+  heatmap:
+    name: Heat map
+    status:
+      key: experimental
+      note:
```
Comment (mattrosno) [inline packages/core/carbon.yml:274]: @theiliad is Heat map experimental? I saw this note in Storybook.

File: packages/core/carbon.yml [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+    name: Stream
+    status: draft
+    type: component
+    tags:
+      - data-visualization
+      - trend
+    framework: design-only
+    platform: web
+    thumbnailPath: /src/thumbnails/stream.svg
+    noIndex: true
```
Comment (mattrosno) [inline packages/core/carbon.yml:479]: I put Steam as `noIndex` so it doesn't show up in the catalogs. It looks grayed out in the legacy site with no click target. This right or wrong?

File: packages/core/carbon.yml (hunk 1 above)
Comment (theiliad) [reply packages/core/carbon.yml:274]: Yes we're still making adjustments to it

File: packages/core/carbon.yml (hunk 2 above)
Comment (theiliad) [reply packages/core/carbon.yml:479]: yeah better to exclude for now

File: packages/core/carbon.yml (hunk 1 above)
Comment (mattrosno) [reply packages/core/carbon.yml:274]: Is this experimental for all? Or just vanilla and stable for others?

File: packages/angular/carbon.yml [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    externalDocsUrl: https://www.carbondesignsystem.com/data-visualization/complex-charts#circle-packs
+    demoLinks:
+      - type: storybook
+        name: Storybook
+        action: link
+        url: https://charts.carbondesignsystem.com/angular/?path=/story/complex-charts-circle-pack
+  combo:
+    status: stable
+    framework: angular
+    externalDocsUrl: https://www.carbondesignsystem.com/data-visualization/getting-started
```
Comment (theiliad) [inline packages/angular/carbon.yml:116]: oh wow that's a gap, but yeah probably the best link for now

**Review by theiliad — APPROVED** [review]
looks good at a quick glance, mainly checked the Angular configs, assuming the rest are almost duplicates

**Comment by theiliad** [comment]
> * [ ]  Do we have descriptions for each chart?

no not yet

> * [ ]  Need tags for (https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#data-visualization-tags)
>   * [ ]  Combo

Probably need a new tag

>   * [ ]  Sparkline

trend

>   * [ ]  Step

I think trends

> * [ ]  Should `Meter (proportional)` be indexed? I see in as top-level in Storybook, but not listed in legacy website index. If so, we need a thumbnail for it.

I think so.

**Review by theiliad — CHANGES_REQUESTED** [review]
could we move `thumbnails` to the root pls?

**Comment by jeanservaas** [comment]
Here are thumbs for:
– step
- combo
- prop. meter
- sparkline

[new_charts.zip](https://github.com/carbon-design-system/carbon-charts/files/8216971/new_charts.zip)

File: packages/angular/carbon.yml (hunk 3 above)
Comment (mattrosno) [reply packages/angular/carbon.yml:116]: Ya, I don't see any cds.com content for Combo, Sparkline, Step. In Carbon Next private beta, all assets need `externalDocsUrl`. So for now, we can point people to the cds.com data viz getting started. In a later platform release, we'll remove the `externalDocsUrl`s when we have detail pages populated.

**Review by theiliad — APPROVED** [review]
good at a quick glance. didn't check the thumbnails, assuming Jeannie did that

**Pull request merged**